src/foam/nanos/crunch/UserCapabilityJunctionExpiredDAO.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * @license
+ * Copyright 2023 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.crunch',
+  name: 'UserCapabilityJunctionExpiredDAO',
+  extends: 'foam.dao.ProxyDAO',
+  flags: ['java'],
+
+  documentation: 'On find, set status to EXPIRED if expired.',
+
+  methods: [
+    {
+      name: 'find_',
+      javaCode: `
+      UserCapabilityJunction ucj = (UserCapabilityJunction) getDelegate().find_(x, id);
+      if ( ucj != null &&
+           ucj.getStatus() == CapabilityJunctionStatus.GRANTED &&
+           ucj.getExpiry() != null &&
+           ! ucj.isInGracePeriod(x) &&
+           ! ucj.isInRenewalPeriod(x) ) {
```

Comment:
if a capability is granted and not yet inRenewalPeriod, wouldn't this also expire that capability? Or does renewal period start as soon as a capability is granted now?

Reply:
Yes, you are correct. I'll address.

Reply:
similar question on this as well
https://github.com/kgrgreer/foam3/pull/2914/files#diff-82329b686948c51f945cdf6a682410139251a426d2cc965be4b8129ee4888eccR235

Reply:
Added isNotYetInRenewalPeriod to capture period between expiry being set and start of renewal period (when renewal period > 0)